parser/parser/cc/token.cc

```diff
@@ -15,13 +15,13 @@ size_t token::start() const {
 }
 
 size_t token::end() const {
-    return _end;
+    return _start + _size;
 }
 
 std::string_view token::view() const {
     return _string;
 }
 
-const std::string &token::asString() const {
-    return _string;
+std::string token::asString() const {
+    return std::string(_string);
 }
```

Comment:
Do we know how often this gets called? From the PR description it sounds like you've already done the homework on bottom line perf, so maybe not any cause for real concern, just noting that we are copying here where we didn't used to.

Reply:
Yeah, some previous PRs helped eliminate the number of places where we would call into this function.

If you look through `parser/Builder.cc` for `asString()`, you can see that we don't call it in that many places.  I think it would help to eliminate it for `BuilderImpl::arg` and maybe one or two other places, but I'm not super-worried about it.